javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java

```diff
@@ -121,37 +113,14 @@ private int considerEnforcingIndentation(NodeText nodeText, int nodeTextIndex) {
      * to the difference (adding and removing the elements provided).
      */
     void apply() {
-        do {
-            if (diffIndex < diffElements.size() && originalIndex >= originalElements.size()) {
-                DifferenceElement diffElement = diffElements.get(diffIndex);
-                if (diffElement instanceof Kept) {
-                    Kept kept = (Kept) diffElement;
+        extractReshuffledDiffElements(diffElements);
```

Comment:
I moved the logic for the Reshuffled element out of the do while loop because it made it more complicated since the Reshuffled elements are "extracted" into additional Addition, Kept and Removed elements which are traversed in the do while loop.

Reply:
It sounds like a smart thing to do